sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ExposureInContactEpiDataSteps.java

```diff
@@ -0,0 +1,278 @@
+import static org.sormas.e2etests.pages.application.contacts.EditEpidemiologicalDataContactPage.EXPOSURE_DETAILS_NEW_ENTRY;
+import static org.sormas.e2etests.pages.application.contacts.EditEpidemiologicalDataContactPage.LARGE_OUTBREAKS_AREA;
+import static org.sormas.e2etests.pages.application.contacts.ExposureNewEntryPage.*;
+
+import cucumber.api.java8.En;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.assertj.core.api.SoftAssertions;
+import org.sormas.e2etests.helpers.WebDriverHelpers;
+import org.sormas.e2etests.pojo.web.ExposureDetails;
+import org.sormas.e2etests.pojo.web.ExposureInvestigation;
+import org.sormas.e2etests.services.ExposureDetailsService;
+import org.sormas.e2etests.services.ExposureInvestigationService;
+import org.sormas.e2etests.state.ApiState;
+
+public class ExposureInContactEpiDataSteps implements En {
+
+  private final WebDriverHelpers webDriverHelpers;
+  private static ExposureInvestigation exposureInvestigationInput;
+  private static ExposureDetails exposureDetailsInput;
+  private String EPIDATA_FOR_LAST_CREATED_CONTACT_URL;
+
+  @Inject
+  public ExposureInContactEpiDataSteps(
+      WebDriverHelpers webDriverHelpers,
+      ExposureInvestigationService exposureInvestigationService,
+      ExposureDetailsService exposureDetailsService,
+      ApiState apiState,
+      @Named("ENVIRONMENT_URL") String environmentUrl) {
+    this.webDriverHelpers = webDriverHelpers;
+
+    When(
+        "I am accessing the Epidemiological tab using of created contact via api",
+        () -> {
+          webDriverHelpers.waitForPageLoaded();
+          EPIDATA_FOR_LAST_CREATED_CONTACT_URL =
+              environmentUrl
+                  + "/sormas-ui/#!contacts/epidata/"
+                  + apiState.getCreatedContact().getUuid();
+          webDriverHelpers.accessWebSite(EPIDATA_FOR_LAST_CREATED_CONTACT_URL);
+          webDriverHelpers.waitForPageLoaded();
+        });
+
+    When(
+        "I check and fill all data",
+        () -> {
+          exposureInvestigationInput =
+              exposureInvestigationService.buildInputExposureInvestigation();
+          exposureIvestigationOnContact(exposureInvestigationInput);
+
+          exposureDetailsInput = exposureDetailsService.buildInputExposureDetails();
+          addNewExposureEntry(exposureDetailsInput);
+          webDriverHelpers.clickOnWebElementBySelector(DONE_BUTTON);
+          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(SAVE_BUTTON);
+          webDriverHelpers.clickOnWebElementBySelector(SAVE_BUTTON);
+          webDriverHelpers.clickOnWebElementBySelector(CONTACT_DATA_SAVED_POPUP);
+          webDriverHelpers.clickOnWebElementBySelector(EXPOSURE_EDIT_BUTTON);
+        });
+
+    Then(
+        "I am checking all data is saved and displayed on edit Exposure page",
+        () -> {
+          ExposureDetails exposureDetailsOutput = getExposureDetailsOutput();
+          SoftAssertions softly = new SoftAssertions();
```

Comment:
inject this object from constructor.
Also, if we're comparing all fields, instead of having these lines, we can resume to one using Truth